sky/services/platform/path_provider.mojom

```diff
@@ -0,0 +1,38 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+[DartPackage="sky_services"]
+module flutter.platform;
+
+/// Returns commonly used locations on the filesystem.
+interface PathProvider {
+  /// Path to the read-only directory that contains the assets packaged with
+  /// the application bundle.
+  ///
+  /// Example(s):
+  ///    iOS: The application bundle directory.
+  AssetsDirectory() => (string path);
+
+  /// Path to the temporary directory on the device. Files in this directory
+  /// may be cleared at any time. This does *not* return a new temporary
+  /// directory. Instead, the caller is responsible for creating
+  /// (and cleaning up) files or directories within this directory. It is
+  /// recommended that applications create subdirectories within this directory
+  /// prefixed by their application identifiers and place their temporary files
```

Comment:
I am not sure its necessary for us to enforce that guarantee. That guarantee may be difficult to enforce.